modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -412,6 +417,26 @@ private void _computeRootFolder() {
 						rootFolder.getName());
 				}
 			}
+
+			if (rootFolder.getGroupId() != _themeDisplay.getScopeGroupId()) {
+				Group group = GroupLocalServiceUtil.getGroup(
+					rootFolder.getGroupId());
+
+				if (group.isDepot()) {
+					List<Long> groupConnectedDepotEntries = ListUtil.toList(
+						DepotEntryServiceUtil.getGroupConnectedDepotEntries(
+							_themeDisplay.getScopeGroupId(), QueryUtil.ALL_POS,
+							QueryUtil.ALL_POS),
+						DepotEntry::getGroupId);
+
+					if (!groupConnectedDepotEntries.contains(
+							rootFolder.getGroupId())) {
+
+						throw new NoSuchFolderException(
+							"{folderId=" + _rootFolderId + "}");
+					}
+				}
+			}
```

Comment:
This logic is duplicated 3 times, and all occurrences are in the same module. Can you move this to an internal util class?